core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1959,47 +1959,58 @@ class SparkContext(config: SparkConf) extends Logging {
             throw new IllegalArgumentException(
               s"Directory ${path} is not allowed for addJar")
           }
-          path
+          Seq(path)
         } catch {
           case NonFatal(e) =>
             logError(s"Failed to add $path to Spark environment", e)
-            null
+            Nil
         }
       } else {
-        path
+        Seq(path)
       }
     }
 
     if (path == null || path.isEmpty) {
       logWarning("null or empty path specified as parameter to addJar")
     } else {
-      val key = if (path.contains("\\") && Utils.isWindows) {
+      val (keys, schema) = if (path.contains("\\") && Utils.isWindows) {
```

Comment:
`schema` -> `scheme` because `scheme` is different from `schema`.

Reply:
If we want to use `plural`, it's `schemes`.

Reply:
> `schema` -> `scheme` because `scheme` is different from `schema`.

Very embarrassed， updated

Reply:
> If we want to use `plural`, it's `schemes`.

Here scheme is singular.